consent/metrics.go

```diff
@@ -0,0 +1,69 @@
+package consent
+
+import (
+	"strconv"
+
+	"github.com/ory/hydra/metrics/prometheus"
+
+	prom "github.com/prometheus/client_golang/prometheus"
+)
+
+func recordAcceptConsentRequest(
+	m *prometheus.MetricsManager,
+	cr *HandledConsentRequest,
+	err error,
+) {
+	m.PrometheusMetrics.ConsentRequests.With(prom.Labels{
+		"error": strconv.FormatBool(err != nil),
+	}).Inc()
+
+	for _, v := range cr.GrantedScope {
+		m.PrometheusMetrics.ConsentRequestScopes.With(prom.Labels{
+			"scope": v,
+			"type":  "granted",
+		}).Inc()
+	}
+
+	for _, v := range cr.GrantedAudience {
+		m.PrometheusMetrics.ConsentRequestAudiences.With(prom.Labels{
+			"audience": v,
+			"type":     "granted",
+		}).Inc()
+	}
+
+	if cr.ConsentRequest != nil {
```

Comment:
Can we send something when this is nil? Under what circumstances would this be nil?

Reply:
I actually am not entirely sure what would cause it to be nil, but when I was doing some basic oauth2 flows using [this](https://www.ory.sh/docs/deprecated/latest/hydra/1-tutorial/), it ended up being `nil` for me and causing a panic.

Reply:
I found out that this field is just never set in the handler, even though the ConsentRequest is retrieved from the database.